### PR TITLE
Feature/condo/DOMA-1210/payments registry

### PR DESCRIPTION
**Pull request by AleX83Xpert**

### CODE REVIEW

File: apps/condo/domains/acquiring/access/ExportPaymentsService.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    if (organizationId) {
+        return await checkOrganizationPermission(user.id, organizationId, permissionToCheck)
+    }
+
+    const organizationWhere = get(where, 'organization')
+    if (!organizationWhere) {
+        return false
+    }
+
+    const [organization] = await find('Organization', organizationWhere)
```
Comment (SavelevMatthew) [inline apps/condo/domains/acquiring/access/ExportPaymentsService.js:27]: I guess `deletedAt: null` should be overwritten here...so you should specify it explicitly...

File: apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const { getById, getByCondition } = require('@core/keystone/schema')
+
+async function canReadPaymentsFilterTemplates ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin) return {}
+
+    return {
+        employee: { organization: { ...queryOrganizationEmployeeFor(user.id) } },
```
Comment (SavelevMatthew) [inline apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js:16]: why related organization in Export access and no related-org logic here?

File: apps/condo/domains/acquiring/components/payments/NoBillingStub.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    const noBillingLabel = intl.formatMessage({ id: 'pages.condo.payments.noBillingLabel' })
+    const noBillingMessage = intl.formatMessage({ id: 'pages.condo.payments.noBillingMessage' })
+    const createBillingLabel = intl.formatMessage({ id: 'pages.condo.payments.createBillingLabel' })
+
+    return (
+        <EmptyListView
+            label={noBillingLabel}
+            message={noBillingMessage}
+            createRoute="/settings"
```
Comment (SavelevMatthew) [inline apps/condo/domains/acquiring/components/payments/NoBillingStub.tsx:16]: `/settings?tab=billing`

File: apps/condo/domains/acquiring/constants.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/**
```
Comment (SavelevMatthew) [inline apps/condo/domains/acquiring/constants.js:1]: Delete this file

File: apps/condo/domains/acquiring/hooks/usePaymentsTableColumns.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                title: accountTitle,
+                key: 'accountNumber',
+                dataIndex: 'accountNumber',
+                width: '120px',
+                render: getTextRender(String(search)),
+            },
+            address: {
+                title: addressTitle,
+                key: 'address',
+                dataIndex: ['receipt', 'property', 'address'],
```
Comment (SavelevMatthew) [inline apps/condo/domains/acquiring/hooks/usePaymentsTableColumns.ts:46]: What if receipt was deleted? I guess then we should display frozenReceipt properties as a fallback...

Logic here are:
1. `row` is raw object
2. Then subObject is taken by dataIndex...in our case `row.receipt.property.address`
3. subObject is passed to `render` function...so it's like `getTextRender(search)(row.receipt.property.address)`

What we can do instead is specify [] or undefinded as dataIndex and then overwrite render:
```
// Can move it to const above columns
const stringSeach = getTextRender(String(search))
render: (obj) => stringSearch(get(obj, ['receipt, ....], obj.frozenReceipt....))
```

File: apps/condo/domains/acquiring/schema/fields/PaymentsFilter.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+const ajv = new Ajv()
+const PaymentsFilterValidator = ajv.compile(PaymentsFilterSchema)
+const validatePaymentsFilter = getValidator(PaymentsFilterValidator)
+
+const PAYMENTS_FILTER_FIELD = {
+    schemaDoc: 'Filter that match the given template',
+    type: Json,
+    extendGraphQLTypes: [PAYMENTS_FILTER_TYPE],
+    graphQLReturnType: PAYMENTS_FILTER_TYPE_NAME,
```
Comment (SavelevMatthew) [inline apps/condo/domains/acquiring/schema/fields/PaymentsFilter.js:39]: How about this fields:
```
graphQLInputType: ...,
graphQLAdminFragment: ...,
```

Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/schema/fields/PaymentsFilter.js:39]: First one needed for create mutations and second is for viewing via admin-panel

**Review by SavelevMatthew — COMMENTED** [review]
Wow, seems like you did really good job here!1!!
I'll approve it tomorrow after building it locally

File: apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    }
+}
+
+async function canManagePaymentsFilterTemplates ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin) return true
+
+    if (operation === 'create') {
+        const employeeForUser = await getByCondition('OrganizationEmployee', {
```
Comment (dkoviazin) [inline apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js:27]:
```js 
{
        isAccepted: true,
        isRejected: false,
}
```

File: apps/condo/domains/acquiring/access/ExportPaymentsService.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/access/ExportPaymentsService.js]: fixed

File: apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js (hunk 2 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js:16]: Here is the same decision as for templates for tickets filtering.
Anyway, this is a business related decision: should we give access to templates for employees from related organizations or not.

File: apps/condo/domains/acquiring/components/payments/NoBillingStub.tsx (hunk 3 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/components/payments/NoBillingStub.tsx]: fixed

File: apps/condo/domains/acquiring/constants.js (hunk 4 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/constants.js]: Done :)

File: apps/condo/domains/acquiring/hooks/usePaymentsTableColumns.ts (hunk 5 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/hooks/usePaymentsTableColumns.ts]: You right. Thanks for detailed explanation!
Fixed.

File: apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js (hunk 2 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js:16]: You right. This security issue fixed.

File: apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js (hunk 7 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/access/PaymentsFilterTemplate.js:27]: Fixed

File: apps/condo/domains/acquiring/schema/fields/PaymentsFilter.js (hunk 6 above)
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/schema/fields/PaymentsFilter.js:46]: Done!

**Review by dkoviazin — APPROVED** [review]

**Pull request merged**